pub/spec/dependabot/pub/update_checker_spec.rb

```diff
@@ -465,6 +468,94 @@
     end
   end
 
+  describe "#lowest_resolvable_security_fix_version" do
+    subject(:lowest_resolvable_security_fix_version) { checker.lowest_resolvable_security_fix_version }
+    let(:dependency_name) { "retry" }
+    let(:security_advisories) do
+      [
+        Dependabot::SecurityAdvisory.new(
+          dependency_name: dependency_name,
+          package_manager: "pub",
+          vulnerable_versions: ["<3.0.0"]
+        )
+      ]
+    end
+
+    context "when a newer non-vulnerable version is available" do
+      # TODO: Implement https://github.com/dependabot/dependabot-core/issues/5391, then flip "highest" to "lowest"
+      it "updates to the highest non-vulnerable version" do
+        is_expected.to eq(Gem::Version.new("3.1.0"))
+      end
+    end
+
+    # TODO should it update indirect deps for security vulnerabilities? I assume Pub has these? example in go_modules/update_checker_spec
```

Comment:
@jonasfj @sigurdm How does `dependency_services` handle bumping transitive deps? I saw a brief mention of them on `pub/README.md` but am unclear if they're ignored or if they get bumped... 

Ultimately I'm trying to figure out if we receive a security vulnerability on a transitive dep, since this PR calls `latest_resolvable_version` will that get applied to transitive deps or skip them?